### PR TITLE
fix(calling): Spark 428312- Media refactor changes

### PR DESCRIPTION
**Pull request by Kesari3008**
# COMPLETES #[SPARK-428312](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-428312)

## This pull request addresses
Refactored media changes in Calling SDK as per WCME refactor effort

## by making the following changes
1. Add the media-helpers module along to ensure we get all the required functions for creating microphone (and camera in future) streams.
2. Accept LocalMicrophoneStream as input for dial and answer functions in call module
3. Pick up the track object from the stream and pass it to RoapMediaConnection. Invoke RoapMediaConnection in the latest manner according to internal-media-core 2.0.0
4. Provide createMicrophoneStream as a static function inside the `Calling` class as we should not need a Calling instance in order to do getMediaStreams on the browser.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [x] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: docs/samples/calling/app.js [hunk 1]
```diff
@@ -569,18 +569,9 @@ function initiateTransfer() {
...
-  try {
-    audioTrack = await callingClient.mediaEngine.Media.createAudioTrack(audio);
-  } catch (e) {
-    console.error(e);
-  }
-
-  localAudioTrack = audioTrack.getMediaStreamTrack();
-  localAudioElem.srcObject = new MediaStream([audioTrack.getMediaStreamTrack()]);
+  localAudioStream  = await calling.createCallingMicrophoneStream({audio: true});
+  // localVideoStream  = await calling.createCallingCameraStream({video: true});
```
Comment (sreenara) [inline docs/samples/calling/app.js:573]: This can be removed.

File: packages/calling/calling/src/CallingClient/calling/call.ts [hunk 2]
```diff
@@ -1859,7 +1859,7 @@ export class Call extends Eventing<CallEventTypes> implements ICall {
    * @param settings.localAudioTrack - MediaStreamTrack.
    * @param settings.debugId - String.
    */
-  private initMediaConnection(settings: {localAudioTrack: MediaStreamTrack; debugId?: string}) {
+  private initMediaConnection(localAudioTrack: MediaStreamTrack, debugId?: string) {
```
Comment (sreenara) [inline packages/calling/calling/src/CallingClient/calling/call.ts:1862]: Are there no changes required in the UT?

File: packages/calling/calling/tsconfig.json [hunk 3]
```diff
@@ -65,6 +65,7 @@
     // "experimentalDecorators": true, /* Enables experimental support for ES7 decorators. */
     // "emitDecoratorMetadata": true, /* Enables experimental support for emitting type metadata for decorators. */
     /* Advanced Options */
+    "stripInternal": true,
```
Comment (sreenara) [inline packages/calling/calling/tsconfig.json:68]: What was the reason for adding this?

**Review by sreenara — CHANGES_REQUESTED** [review]
Please see the comments. Also correct the PR description according to the final changes.
Can we also have a change in the [Call confluence doc](https://confluence-eng-gpk2.cisco.com/conf/pages/viewpage.action?pageId=386033589)

File: packages/calling/calling/tsconfig.json (hunk 3 above)
Comment (Kesari3008) [reply packages/calling/calling/tsconfig.json:68]: To fix the import with media-helper, it was added. Not needed anymore

File: docs/samples/calling/app.js (hunk 1 above)
Comment (Kesari3008) [reply docs/samples/calling/app.js:573]: removed

**Review by sreenara — APPROVED** [review]
Changes look good.

**Pull request merged**